rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/CloudProviderBakeHandler.groovy

```diff
@@ -140,7 +141,21 @@ abstract class CloudProviderBakeHandler {
     def virtualizationSettings = findVirtualizationSettings(region, bakeRequest)
 
     BakeOptions.Selected selectedOptions = new BakeOptions.Selected(baseImage: findBaseImage(bakeRequest))
-    def (imageName, appVersionStr, packagesParameter) = imageNameFactory.deriveImageNameAndAppVersion(bakeRequest, selectedOptions)
+
+    List<String> packageNameList = bakeRequest?.package_name?.tokenize(" ")
+    def appVersionStr
+
+    def osPackages = packageNameList.collect { name ->
+      PackageNameConverter.buildOsPackageName(selectedOptions.baseImage.packageType, name)
+    }
+
+    if (!osPackages.empty) {
+     appVersionStr = imageNameFactory.buildAppVersionStr(bakeRequest, osPackages.first())
```

Comment:
Initially it made sense since I'm basically just refactoring out the existing behaviour which only uses the first package. But looking at it again I think it's a good idea to send inn the entire list as it might be useful for someone else, and it puts the logic of making this choice where it belongs; within the method which can be overridden.